presto-connector-tests/src/main/java/com/facebook/presto/connector/unittest/BaseMetadataTest.java

```diff
@@ -0,0 +1,243 @@
+                    metadata -> metadata.dropTable(session, metadata.getTableHandle(session, schemaTableName))));
+        }
+    }
+
+    default List<String> schemaNamesOf(SchemaTableName... schemaTableNames)
+    {
+        return Arrays.stream(schemaTableNames)
+                .map(SchemaTableName::getSchemaName)
+                .collect(toImmutableList());
+    }
+
+    default List<String> schemaNamesOf(List<ConnectorTableMetadata> tables)
+    {
+        return tables.stream()
+                .map(ConnectorTableMetadata::getTable)
+                .map(SchemaTableName::getSchemaName)
+                .distinct()
+                .collect(toImmutableList());
+    }
+
+    default SchemaTablePrefix prefixOfSchemaName(SchemaTableName schemaTableName)
+    {
+        return new SchemaTablePrefix(schemaTableName.getSchemaName());
+    }
+
+    default SchemaTablePrefix prefixOf(SchemaTableName schemaTableName)
+    {
+        return new SchemaTablePrefix(schemaTableName.getSchemaName(), schemaTableName.getTableName());
+    }
+
+    default void run(List<Consumer<ConnectorMetadata>> consumers)
+    {
+        consumers.forEach(this::withMetadata);
+    }
+
+    default void withMetadata(Consumer<ConnectorMetadata> consumer)
+    {
+        Connector connector = getConnector();
+        ConnectorTransactionHandle transaction = connector.beginTransaction(IsolationLevel.READ_UNCOMMITTED, true);
+        ConnectorMetadata metadata = connector.getMetadata(transaction);
+        consumer.accept(metadata);
+        connector.commit(transaction);
+    }
+
+    class AllCloser
+            implements AutoCloseable
+    {
+        List<AutoCloseable> items;
+
+        public AllCloser(List<AutoCloseable> items)
+        {
+            this.items = items;
+        }
+
+        static AllCloser of(AutoCloseable... elements)
+        {
+            return new AllCloser(ImmutableList.copyOf(elements));
+        }
+
+        @Override
+        public void close()
+                throws Exception
+        {
+            for (int i = items.size() - 1; i >= 0; --i) {
+                items.get(i).close();
+            }
+        }
+    }
+
+    class Schema
+            implements AutoCloseable
+    {
+        private final BaseMetadataTest test;
+        private final ConnectorSession session;
+        private final String name;
+
+        public Schema(BaseMetadataTest test, ConnectorSession session, String name)
+        {
+            this.test = test;
+            this.session = session;
+            this.name = name;
+
+            test.withMetadata(metadata -> metadata.createSchema(session, name, ImmutableMap.of()));
+        }
+
+        @Override
+        public void close()
+                throws Exception
+        {
+            test.withMetadata(metadata -> metadata.dropSchema(session, name));
+        }
+    }
+
+    default List<AutoCloseable> withSchemas(ConnectorSession session, List<String> schemaNames)
```

Comment:
same suggestion as for `withTables`. Is there a benefit from exposing `AutoClosable`s to caller. Are there cases when it would not be immediatelly wrapped in `try`. If that is not the case I suggest to use `Runnable` as last argument and use lambdas when it is used.